packages/fxa-settings/src/lib/hooks/useTotpSetup/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { createContext, useContext } from 'react';
+import { useAsync } from 'react-async-hook';
+import { useAccount, useSession } from '../../../models';
+import { TotpInfo } from '../../types';
+
+export const TotpSetupContext = createContext<
+  | {
+      totpInfo: TotpInfo;
+      loading: boolean;
+      error: Error | null;
+    }
+  | undefined
+>(undefined);
+
+export const useTotpSetup = () => {
```

Comment:
Did you read through [Before You Use Context](https://legacy.reactjs.org/docs/context.html#before-you-use-context)? Just double checking since I am not sure if it's a better solution than a parent component to hold the state here, but I also am happy to revisit later since this is a long PR and I'm here at the end of my initial pass lol.

Reply:
Chose a hook here because we'll be reusing this function in at least two other places in upcoming epics/tickets: for inline totp setup and for changing 2fa app.

Reply:
Using a shared hook is fine, I just wasn't sure about a new `Context` since we've wanted to use them sparingly when possible. I think I'm maybe still not understanding why we need it, is it just for stories/tests? If so we could just mock the return value... Will we at least refactor away using `useAccount();` if we intend on using this for inline TOTP setup?

Reply:
Ditto from above, I'd intended to remove context. Next push will have the change